### PR TITLE
Added new target support for dataview object

### PR DESCRIPTION
**Pull request by kisbg**
JerryScript-DCO-1.0-Signed-off-by: bence gabor kis kisbg@inf.u-szeged.hu

### CODE REVIEW

File: jerry-core/ecma/operations/ecma-dataview-object.c [hunk 1]
```diff
@@ -45,7 +47,10 @@ ecma_op_dataview_create (const ecma_value_t *arguments_list_p, /**< arguments li
                          ecma_length_t arguments_list_len) /**< number of arguments */
 {
   JERRY_ASSERT (arguments_list_len == 0 || arguments_list_p != NULL);
-
+  if (!JERRY_CONTEXT (current_new_target))
```
Comment (rerobika) [inline jerry-core/ecma/operations/ecma-dataview-object.c:50]: That should have been an assert.

File: jerry-core/ecma/operations/ecma-dataview-object.c [hunk 2]
```diff
@@ -127,7 +132,9 @@ ecma_op_dataview_create (const ecma_value_t *arguments_list_p, /**< arguments li
   }
 
   /* 13. */
-  ecma_object_t *object_p = ecma_create_object (ecma_builtin_get (ECMA_BUILTIN_ID_DATAVIEW_PROTOTYPE),
+  ecma_object_t *prototype_obj_p = ecma_op_get_prototype_from_constructor (JERRY_CONTEXT (current_new_target),
```
Comment (rerobika) [inline jerry-core/ecma/operations/ecma-dataview-object.c:135]: This operation may raises error.

File: tests/jerry/es2015/new-target-for-dataview.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+var buffer = new ArrayBuffer (16);
+try {
+ Reflect.construct (DataView, buffer);
+ assert (false);
+} catch (e) {
+ assert (e instanceof TypeError);
+}
+
+try {
+ Reflect.construct (DataView);
```
Comment (rerobika) [inline tests/jerry/es2015/new-target-for-dataview.js:25]: I think this tc is not related to this feature. I'd add a tc when the 3rd argument is present.

File: tests/jerry/es2015/new-target-for-dataview.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+//
+var buffer = new ArrayBuffer (16);
+try {
+ Reflect.construct (DataView, buffer);
```
Comment (rerobika) [inline tests/jerry/es2015/new-target-for-dataview.js:18]: Arguments must be passed in an array, like `[buffer]`

File: jerry-core/api/jerry.c [hunk 5]
```diff
@@ -3931,8 +3931,15 @@ jerry_create_dataview (const jerry_value_t array_buffer, /**< arraybuffer to cre
     ecma_make_uint32_value (byte_offset),
     ecma_make_uint32_value (byte_length)
   };
+  ecma_object_t * old_new_target_p = JERRY_CONTEXT (current_new_target);
```
Comment (rerobika) [inline jerry-core/api/jerry.c:3934]: no space after `*`

File: jerry-core/api/jerry.c [hunk 6]
```diff
@@ -3928,9 +3928,16 @@ jerry_create_dataview (const jerry_value_t array_buffer, /**< arraybuffer to cre
...
+  {
+    JERRY_CONTEXT (current_new_target) = ecma_builtin_get (ECMA_BUILTIN_ID_DATAVIEW);
+  }
 
-  return jerry_return (ecma_op_dataview_create (arguments_p, 3));
-#else /* !ENABLED (JERRY_BUILTIN_DATAVIEW) */
+  ecma_value_t dataview_value = ecma_op_dataview_create (arguments_p, 3);
+  JERRY_CONTEXT (current_new_target) = old_new_target_p;
+  return jerry_return (dataview_value);
+#else /* !ENABLED (JERRY_ES2015_BUILTIN_DATAVIEW) */
```
Comment (dbatyai) [inline jerry-core/api/jerry.c:3940]: Comment is incorrect.
